src/spectrum.jl

```diff
@@ -0,0 +1,177 @@
+export spectrum, spectrum_correlation_fft
+export SpectrumSolver, ExponentialSeries, PseudoInverse
+
+abstract type SpectrumSolver end
+
+@doc raw"""
+    ExponentialSeries(; tol = 1e-14, calc_steadystate = false)
+
+A solver which solves [`spectrum`](@ref) by finding the eigen decomposition of the Liouvillian [`SuperOperator`](@ref) and calculate the exponential series.
+"""
+struct ExponentialSeries{T<:Real,CALC_SS} <: SpectrumSolver
+    tol::T
+    ExponentialSeries(tol::T, calc_steadystate::Bool = false) where {T} = new{T,calc_steadystate}(tol)
+end
+
+ExponentialSeries(; tol = 1e-14, calc_steadystate = false) = ExponentialSeries(tol, calc_steadystate)
+
+@doc raw"""
+    PseudoInverse(; alg::SciMLLinearSolveAlgorithm = KrylovJL_GMRES())
+
+A solver which solves [`spectrum`](@ref) by finding the inverse of Liouvillian [`SuperOperator`](@ref) using the `alg`orithms given in [`LinearSolve.jl`](https://docs.sciml.ai/LinearSolve/stable/).
+"""
+struct PseudoInverse{MT<:SciMLLinearSolveAlgorithm} <: SpectrumSolver
+    alg::MT
+end
+
+PseudoInverse(; alg::SciMLLinearSolveAlgorithm = KrylovJL_GMRES()) = PseudoInverse(alg)
+
+@doc raw"""
+    spectrum(H::QuantumObject,
+        ωlist::AbstractVector,
+        c_ops::Union{Nothing,AbstractVector,Tuple},
+        A::QuantumObject{<:AbstractArray{T2},OperatorQuantumObject},
+        B::QuantumObject{<:AbstractArray{T3},OperatorQuantumObject};
+        solver::SpectrumSolver=ExponentialSeries(),
+        kwargs...)
+
+Calculate the spectrum of the correlation function
+
+```math
+S(\omega) = \int_{-\infty}^\infty \lim_{t \rightarrow \infty} \left\langle \hat{A}(t + \tau) \hat{B}(t) \right\rangle e^{-i \omega \tau} d \tau
+```
+
+See also the following list for `SpectrumSolver` docstrings:
+- [`ExponentialSeries`](@ref)
+- [`PseudoInverse`](@ref)
+"""
+function spectrum(
+    H::QuantumObject{MT1,HOpType},
+    ωlist::AbstractVector,
+    c_ops::Union{Nothing,AbstractVector,Tuple},
+    A::QuantumObject{<:AbstractArray{T2},OperatorQuantumObject},
+    B::QuantumObject{<:AbstractArray{T3},OperatorQuantumObject};
+    solver::SpectrumSolver = ExponentialSeries(),
+    kwargs...,
+) where {MT1<:AbstractMatrix,T2,T3,HOpType<:Union{OperatorQuantumObject,SuperOperatorQuantumObject}}
+    return _spectrum(liouvillian(H, c_ops), ωlist, A, B, solver; kwargs...)
+end
+
+function _spectrum_get_rates_vecs_ss(L, solver::ExponentialSeries{T,true}) where {T}
+    result = eigen(L)
+    rates, vecs = result.values, result.vectors
+
+    return rates, vecs, steadystate(L).data
```

Comment:
I don't think we need this. The steady state is already computer in `eigen(L)`, so this is unnecessary I think.

Reply:
I just copied this part from the original file. lol

The `ExponentialSeries` provides two methods depend on the parameter `calc_steadystate = false`.

I think this part is for `calc_steadystate = true` ?

Reply:
from the data of `codecov`, our runtests didn't cover this part.

What do you think, do you want to keep this functionality ?

Reply:
Aaaah sorry! Now I remember we we do need this. Sometimes the steady state estimation is better when calling `steadystate` rather that diagonalizing the liouvillian